tree/dataframe/inc/ROOT/RResultHandle.hxx

```diff
@@ -0,0 +1,64 @@
+// Author: Stefan Wunsch, Enrico Guiraud CERN  09/2020
+
+/*************************************************************************
+ * Copyright (C) 1995-2020, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT_RRESULTHANDLE
+#define ROOT_RRESULTHANDLE
+
+#include "ROOT/RResultPtr.hxx"
+#include "ROOT/RDF/RLoopManager.hxx"
+#include "ROOT/RDF/RActionBase.hxx"
+
+namespace ROOT {
+namespace RDF {
+
+namespace Internal {
+
+// Cast the pointer and trigger the event loop
+template <class T>
+void CastAndTriggerRun(void* ptr)
+{
+   // The call to GetPtr triggers the event loop
+   static_cast<RResultPtr<T>*>(ptr)->GetPtr();
+}
+
+} // namespace Internal
+
+class RResultHandle {
+   void *fResultPtr; // Pointer to the actual RResultPtr
+   using Trigger_t = void (*) (void*);
+   Trigger_t fTriggerRun; // Function pointer to function which triggers the event loop of the RResultPtr
+   ROOT::Detail::RDF::RLoopManager* fResultPtrLoopManager; // Pointer to the loop manager of the RResultPtr
```

Comment:
Copy over essential information from the `RResultPtr` (shared ptrs to action and actual value)